src/Modules/OrchardCore.Commerce/Services/PriceProvider.cs

```diff
@@ -55,6 +56,22 @@ public async Task<ShoppingCartItem> AddPriceAsync(ShoppingCartItem item)
         return item;
     }
 
+    public async Task<bool> IsApplicableAsync(IList<ShoppingCartItem> items)
+    {
+        var skuProducts = await PriceProviderHelpers.GetSkuProductsAsync(items, _productService);
+
+        return items
+            .All(item =>
+            {
+                if (skuProducts.TryGetValue(item.ProductSku, out var productPart))
+                {
+                    return productPart.ContentItem.OfType<PricePart>().Any();
+                }
+
+                return false;
+            });
```

Comment:
How about saving a few lines?
```suggestion
        return items.Select(item => skuProducts.TryGetValue(item.ProductSku, out var productPart)
            ? AddPriceToShoppingCartItem(item, productPart)
            : item);
```